src/main/scala/is/hail/stats/HWECombiner.scala

```diff
@@ -14,13 +16,17 @@ class HWECombiner extends Serializable {
   var nHet = 0
   var nHomVar = 0
 
-  def merge(gt: Call): HWECombiner = {
-    if (Call.isHomRef(gt))
-      nHomRef += 1
-    else if (Call.isHet(gt))
-      nHet += 1
-    else if (Call.isHomVar(gt))
-      nHomVar += 1
+  def merge(c: Call): HWECombiner = {
+    (Call.ploidy(c): @switch) match {
+      case 2 =>
+        if (Call.isHomRef(c))
+          nHomRef += 1
+        else if (Call.isHet(c))
+          nHet += 1
+        else if (Call.isHomVar(c))
+          nHomVar += 1
+      case _ =>
```

Comment:
Are we sure we want to do nothing here?  (I know that's the behavior of the old code, but ...).  HWE should require biallelic, right?  So every call should be one of the above?

Reply:
We've never asserted that the genotypes had to be biallelic for the aggregator. You could actually do this which would make no sense `ds.annotate_cols(hwe=agg.hardy_weinberg(ds.GT))`.

The case I was addressing was so haploid samples (and eventually N-ploid samples) aren't incorporated into the statistic because it doesn't make sense. I added a warning to the aggregator docs.

I asked Ben about a general HWE equation to incorporate all ploidies and number of alleles. He didn't think there was an exact formula that encompassed everything, but it's possible we could do a maximum likelihood estimation of some sort.